### PR TITLE
feat: add only-audio-mode support

### PR DESCRIPTION
**Pull request by papamishka911**
Need to add audio-only mode support for call without video tracks in mediastream.

### CODE REVIEW

File: src/SipConnector.ts [hunk 1]
```diff
@@ -937,21 +951,35 @@ export default class SipConnector {
...
   };
 
-  _generateStream(videoTrack, audioTrack) {
-    const { id } = videoTrack;
+  _generateStream({
+    videoTrack,
+    audioTrack,
+  }: {
+    videoTrack?: MediaStreamTrack;
+    audioTrack?: MediaStreamTrack;
```
Comment (Krivega) [inline src/SipConnector.ts:959]: in order to get away from unnecessary conditions, you can create a separate method for this

File: src/SipConnector.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ const BUSY_HERE_STATUS_CODE = 486;
 const REQUEST_TERMINATED_STATUS_CODE = 487;
 const ORIGINATOR_LOCAL = 'local';
 const ORIGINATOR_REMOTE = 'remote';
+const KIND_VIDEO_REMOTE_TRACK = 'video';
```
Comment (Krivega) [inline src/SipConnector.ts:71]: it's not required

Comment (papamishka911) [reply src/SipConnector.ts]: removed

File: src/SipConnector.ts (hunk 1 above)
Comment (papamishka911) [reply src/SipConnector.ts]: New method for generating audio stream has added

File: src/SipConnector.ts [hunk 3]
```diff
@@ -937,8 +950,16 @@ export default class SipConnector {
     });
   };
 
-  _generateStream(videoTrack, audioTrack) {
-    const { id } = videoTrack;
+  _generateStream(
+    videoTrack: MediaStreamTrack,
+    audioTrack?: MediaStreamTrack
+  ): MediaStream | undefined {
+    const id = videoTrack?.id || audioTrack?.id;
```
Comment (Krivega) [inline src/SipConnector.ts:957]: why this changed?

Comment (papamishka911) [reply src/SipConnector.ts]: restored (was changed for creating audio stream with no video track)

File: src/SipConnector.ts [hunk 4]
```diff
@@ -937,8 +950,16 @@ export default class SipConnector {
     });
   };
 
-  _generateStream(videoTrack, audioTrack) {
-    const { id } = videoTrack;
+  _generateStream(
+    videoTrack: MediaStreamTrack,
+    audioTrack?: MediaStreamTrack
+  ): MediaStream | undefined {
```
Comment (Krivega) [inline src/SipConnector.ts:956]: ): MediaStream  {

File: src/SipConnector.ts [hunk 5]
```diff
@@ -937,8 +950,16 @@ export default class SipConnector {
...
+    const id = videoTrack?.id;
```
Comment (Krivega) [inline src/SipConnector.ts:957]: id is exist

File: src/SipConnector.ts [hunk 6]
```diff
@@ -969,12 +1002,26 @@ export default class SipConnector {
...
+      }
     }, []);
 
     return remoteStreams;
   }
 
+  _generateAudioStreams(remoteTracks: MediaStreamTrack[]): MediaStream[] {
+    const remoteStreams: MediaStream[] = [];
+
+    remoteTracks.forEach((audioTrack) => {
```
Comment (Krivega) [inline src/SipConnector.ts:1016]: use map

File: src/SipConnector.ts [hunk 7]
```diff
@@ -951,7 +972,19 @@ export default class SipConnector {
...
+
+    this._remoteStreams[id] = remoteStream;
+
+    return remoteStream;
+  }
+
+  _generateStreams(remoteTracks: MediaStreamTrack[]): MediaStream[] {
     const remoteStreams: MediaStream[] = [];
 
     remoteTracks.forEach((track, index) => {
```
Comment (Krivega) [inline src/SipConnector.ts:990]: use map this to

File: src/SipConnector.ts (hunk 4 above)
Comment (papamishka911) [reply src/SipConnector.ts]: done

File: src/SipConnector.ts (hunk 5 above)
Comment (papamishka911) [reply src/SipConnector.ts]: done

File: src/SipConnector.ts (hunk 6 above)
Comment (papamishka911) [reply src/SipConnector.ts]: done

File: src/SipConnector.ts (hunk 7 above)
Comment (papamishka911) [reply src/SipConnector.ts]: also done

**Pull request merged**